mongodbatlas/resource_mongodbatlas_cluster.go

```diff
@@ -1535,10 +1535,11 @@ func flattenPrivateEndpoint(privateEndpoints []matlas.PrivateEndpoint) []map[str
 	endpoints := make([]map[string]interface{}, 0)
 	for _, endpoint := range privateEndpoints {
 		endpoints = append(endpoints, map[string]interface{}{
-			"connection_string":     endpoint.ConnectionString,
-			"srv_connection_string": endpoint.SRVConnectionString,
-			"endpoints":             flattenEndpoints(endpoint.Endpoints),
-			"type":                  endpoint.Type,
+			"connection_string":                     endpoint.ConnectionString,
+			"srv_connection_string":                 endpoint.SRVConnectionString,
+			"srv_shard_optimized_connection_string": endpoint.SRVShardOptimizedConnectionString,
+			"endpoints":                             flattenEndpoints(endpoint.Endpoints),
+			"type":                                  endpoint.Type,
```

Comment:
Do you need to change `mongodbatlas_advanced_cluster` too?

Reply:
_advanced_cluster resources use the same methods to read/flatten connectionStrings.PrivateEndpoint so these changes will propagate to `mongodbatlas_advanced_cluster` as well

Reply:
Thanks for the clarification!